src/adapters/index.ts

```diff
@@ -6,7 +6,7 @@ export * from "./telegram";
 
 export const createAdapters = (config: BotConfig): Adapters => {
   return {
-    supabase: supabase(config.supabase.url, config.supabase.key),
-    telegram: new Telegraf(config.telegram.token).telegram,
+    supabase: supabase(config?.supabase?.url ?? process.env.SUPABASE_URL, config?.supabase?.key ?? process.env.SUPABASE_KEY),
+    telegram: new Telegraf(config?.telegram?.token ?? process.env.TELEGRAM_BOT_TOKEN).telegram,
```

Comment:
It's not a good idea to put the env variables here. `env` variables should be read in [config.ts](https://github.com/ubiquity/ubiquibot/blob/development/src/bindings/config.ts)

Reply:
bumping @whilefoo here. lets follow up the general process to resolve the conversation. you should at least leave a comment.

Reply:
Even if config separation resolves the problem finally, we should fix it here as well.

Reply:
How can we fix it here? Config is not available if it's invalid

Reply:
yeah I mean if the config is not available here(maybe null), we don't need to put env variables for adapters because the bot instance would not work. If we have to forcely make the supabase working for logging, we can allow it only for supabase adapter but not for telegram adapter. Ideally we should remove it from supabase adapter as well in the future.

Reply:
ok lets make it simple. suggested code changes